### PR TITLE
opamFile extensions (x- fields) enhancements

### PR DESCRIPTION
**Pull request by hannesm**
this PR consists of two commits:
- add `remove_extension : t -> string -> t`
- change behaviour of `to_string_with_preserved_format` to not treat `x-` specially. I'm not sure why the code was around, is there a test / example for the previous behaviour @AltGr?

### CODE REVIEW

File: src/format/opamFile.ml [hunk 1]
```diff
@@ -823,10 +823,7 @@ module Syntax = struct
                    in
                    OpamPrinter.items [f] :: strs
                  with Not_found -> strs
-             with Not_found | OpamPp.Bad_format _ ->
-               if OpamStd.String.starts_with ~prefix:"x-" name then
-                 field_str (`Var name) :: strs
-               else strs)
+             with Not_found | OpamPp.Bad_format _ -> strs
```
Comment (rjbou) [inline src/format/opamFile.ml]:
```suggestion
             with Not_found | OpamPp.Bad_format _ -> strs)
```

**Comment by rjbou** [comment]
It's to keep `x-` fields at their place, instead of moving them if they are in `rem`. There is no test for that function(ality), only manual ones ;).
Why do you want to remove the `x-` fields special case ?

**Comment by hannesm** [comment]
@rjbou because otherwise even if I remove `x-` fields from the OPAM.t, they are retained in the output file when using `to_string_with_preserved_format`

File: src/format/opamFile.ml [hunk 2]
```diff
@@ -823,10 +823,7 @@ module Syntax = struct
                    in
                    OpamPrinter.items [f] :: strs
                  with Not_found -> strs
-             with Not_found | OpamPp.Bad_format _ ->
-               if OpamStd.String.starts_with ~prefix:"x-" name then
-                 field_str (`Var name) :: strs
-               else strs)
+             with Not_found | OpamPp.Bad_format _ -> strs)
```
Comment (rjbou) [inline src/format/opamFile.ml]:
```suggestion
             with Not_found | OpamPp.Bad_format _ ->
               if OpamStd.String.starts_with ~prefix:"x-" name &&
                  List.find_opt (fun i -> it_ident i = `Var name)
                    syn_t.file_contents <> None then
                 field_str (`Var name) :: strs
               else strs)
```

**Comment by rjbou** [comment]
In that case, better check is `x-` field is present in the given opamfile.

**Comment by hannesm** [comment]
thanks @rjbou -- that is fine with me as well (see above, I just don't had any idea why the code was around, but thanks for explaining it), and I committed your patch :D

**Comment by rjbou** [comment]
Cleaning the branch (squash of the 3 last commits) and it's a good to merge for me.

**Comment by hannesm** [comment]
thanks, I squashed and fixed the commit message

**Pull request merged**